sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -415,9 +415,22 @@ case class AlterTableChangeColumnCommand(
     Seq.empty[Row]
   }
 
+  private def verifyNonPartitionColumn(
+      table: CatalogTable, name: String, resolver: Resolver): Unit = {
+    table.partitionSchema.fields.collectFirst {
+        case field if resolver(field.name, name) => field
+    } match {
+      case Some(_) =>
+        throw QueryCompilationErrors.alterTableChangeColumnNotSupportedForPartitionColumn(
+          table.qualifiedName,
+          columnName)
+      case None =>
+    }
+  }
+
   // Find the origin column from schema by column name, throw an AnalysisException if the column
   // reference is invalid.
-  private def findColumnByName(
```

Comment:
shall we add `def findColumnByName(...): Option[StructField]`, so that we can use it to find both partition and data columns?